### DISCUSSION TITLE
No viable ESLint config `languageOptions.sourceType` option for `.cts` modules

### DISCUSSION THREAD

**Discussion by jaydenseric** [eslint/eslint#18798] [Q&A]
As far as I can tell, there is no viable ESLint config `languageOptions.sourceType` option for `.cts` modules.

TypeScript compiles these modules to CJS modules, so the CommonJS globals should be available, but the syntax should only be `import` and not `require`. While `import` syntax should be used, `import.meta` should not be used, or else when Node.js runs the compiled CJS module at runtime it will error with `SyntaxError: Cannot use 'import.meta' outside a module`.

If you configure `.cts` modules to have `languageOptions.sourceType` of `module`, you can't use the CJS globals and you can incorrectly use `import.meta`. But if you configure it to `commonjs`, then you can't use `import` syntax.

What we need is a new `languageOptions.sourceType` option representing import syntax, but CJS runtime.

In the meantime, what is the least worst option, for `.cts` modules? I'm thinking maybe setting it to `module`, and then custom setting up rules to ban the `import.meta` syntax, and to manually configure the globals for CJS.

**Comment by nzakas** [comment]
You're talking about the default settings for `languageOptions.sourceType`, which is not intended for use with TypeScript.

You'll need to talk to the typescript-eslint folks for help with this:
https://typescript-eslint.io/
